luarules/gadgets/unit_infestor_spawn.lua

```diff
@@ -19,23 +19,13 @@ end
 
 
 
-local isBuilding = {}
-local isInfestor = {}
-for udid, ud in pairs(UnitDefs) do
-	if string.find(ud.name, 'leginfestor') then
-		isInfestor[udid] = true
-		Spring.Echo(udid)
-	end
-	if ud.isBuilding then
-		isBuilding[udid] = true
-	end
-end
-
+local infestorId = UnitDefNames.leginfestor.id
```

Comment:
I should have mentioned this when we were talking about things earlier, but Floris is correct that this will crash if legion units aren't loaded. Best to do something like:
```
local infestorId
local infestor = UnitDefNames.leginfestor
if infestor then
    infestorId = infestor.id
else
   gadgetHandler:RemoveGadget(self)
end
```

Reply:
Best to make it a customparam

Reply:
already resolved, but gadget removal is a nice idea which I've integrated

side note that legion disabling seems not to be present in live, for whatever reason; it's perfectly possible to still spawn legion units without the option toggled.